src/main/java/com/microsoft/sqlserver/jdbc/TimeoutPoller.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Thread that runs in the background while the mssql driver is used that can timeout TDSCommands Checks all registered
+ * commands every second to see if they can be interrupted
+ */
+public final class TimeoutPoller implements Runnable {
+    private List<TimeoutCommand> timeoutCommands = new ArrayList<>();
+    final static Logger logger = Logger.getLogger("com.microsoft.sqlserver.jdbc.internals.TDS.Command");

```

Comment:
@cheenamalhotra , please confirm if we need a new logger name here.

Reply:
Logger class should change here to TimeoutPoller.

Reply:
done